### PR TITLE
fixes the --set option in configure.py

### PR DESCRIPTION
**Pull request by shourya5**
closes issue #107049 
This works by appending the CLI arguments to the end of the config.toml file. I am not sure this would work for all scenarios, but a good starting point regardless.

### CODE REVIEW

**Comment by jyn514** [comment]
I don't have time for reviews right now, sorry.

r? @Mark-Simulacrum

**Comment by Mark-Simulacrum** [comment]
This seems OK, but CI needs to be fixed before this can be merged.

@rustbot author

**Comment by shourya5** [comment]
I think its a whitespace issue, I will fix it right away

**Comment by shourya5** [comment]
Does this look okay @Mark-Simulacrum

**Comment by jyn514** [comment]
@shourya5 it can take a little while for reviews. If you haven't heard back in 10 days feel free to ping me or Mark.

**Comment by jyn514** [comment]
You can use `@rustbot ready` to switch the labels like I did just now.

File: src/bootstrap/configure.py [hunk 1]
```diff
@@ -158,8 +183,10 @@ def v(*args):
 
 # Many of these are saved below during the "writing configuration" step
 # (others are conditionally saved).
-o("manage-submodules", "build.submodules", "let the build manage the git submodules")
-o("full-bootstrap", "build.full-bootstrap", "build three compilers instead of two (not recommended except for testing reproducible builds)")
+o("manage-submodules", "build.submodules",
+  "let the build manage the git submodules")
+o("full-bootstrap", "build.full-bootstrap",
+  "build three compilers instead of two (not recommended except for testing reproducible builds)")
```
Comment (Mark-Simulacrum) [inline src/bootstrap/configure.py:189]: Can we undo the re-formatting of this file?

**Pull request closed without merging**